canonical/ubuntu.com
# Update marketo with new user preferences

## Done

- Create a python function to handle POST requests to Marketo
- Allow fetch and update of 'canonical updates opt-in' field
- Move modal trapping into reusable component

## QA

- Run the project on local or using the demo link, see [example url](https://pastebin.canonical.com/p/8JkdySRcjM/)
- Check the fields that are checked, are the same as on the [user on marketo](https://pastebin.canonical.com/p/79WVzZw2tF/)
- Update some fields and submit, check that the page reloads with the correct information and that it has updated on Marketo
- Check the modal still works as expected (except the 'unsubscribe' button)

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-232
